Feature/switch solutions (#843)
MapsPeople/docs: src/content/data/various/switch-solutions.md

    //Initiates the MapControl
    mMapControl.init(miError -> {
        //Orient your map to where you need data to be shown. This could be done by getting the default venue through MapsIndoors and panning the camera there
    });
}
```

</mi-tab-panel>
<mi-tab-panel id="android-kotlin">

```kotlin
override fun onCreate(savedInstanceState: Bundle?) {
    ...
    MapsIndoors.initialize(applicationContext, "YOUR_MAPSINDOORS_API_KEY")
    MapsIndoors.setGoogleAPIKey(“YOUR_GOOGLE_API_KEY”)
    mapFragment.view?.let {
        mapView = it
    }
    mapFragment.getMapAsync(this)
    ...
}
override fun onMapReady(googleMap: GoogleMap) {
    mMap = googleMap
    initMapControl(mMapView)
}
fun initMapControl(view: View) {
    //Creates a new instance of MapControl
    mMapControl = MapControl(this)
    //Sets the Google map object and the map view to the MapControl
    mMapControl.setGoogleMap(mMap, view)
    //Initiates the MapControl
    mMapControl.init { miError: MIError? ->
        //Orient your map to where you need data to be shown. This could be done by getting the default venue through MapsIndoors and panning the camera there
    }
}
```

</mi-tabs>

### Switching Solutions

> Since you took the time to set up your Solution "properly" previously, switching Solutions to a different one is as simple as changing the active API key using `setAPIKey()`, along with ensuring that `mMapControl` doesn't retain any uneccesary information from the previously active Solution, which could cause conflicts.

Reviewer: Same as comment above, whats not properly? MapControl wont retain anything of conflict. However a developer should be aware of previous data received from MapsIndoors might be nulled.

> We recommend initialising your own function to call in the future for this purpose, like the example here with `switchSolution()`:

Reviewer 2: ```suggestion
We recommend creating your own function to call in the future for this purpose, like the example here with `switchSolution()`:
```

<mi-tabs>
<mi-tab label="Java" tab-for="android-java"></mi-tab>
<mi-tab label="Kotlin" tab-for="android-kotlin"></mi-tab>
<mi-tab-panel id="android-java">

```java
protected void switchSolution() {
    //setApiKey returns a boolean, if MapsIndoors is in a state where it is possible to call setApiKey. This does not validate your api key.
    if(MapsIndoors.setAPIKey("YOUR_SECONDARY_API_KEY")) {
        MapsIndoors.setGoogleAPIKey(“YOUR_GOOGLE_API_KEY”);
        //If you already have a mapControl running this needs to be destroyed
        mMapControl.onDestroy();
        mapFragment.getMapAsync(this);  
    }
}
```

</mi-tab-panel>
<mi-tab-panel id="android-kotlin">

```kotlin
private fun switchSolution() {
    //setApiKey returns a boolean, if MapsIndoors is in a state where it is possible to call setApiKey. This does not validate your api key.
    if (MapsIndoors.setAPIKey("YOUR_SECONDARY_API_KEY")) {
        MapsIndoors.setGoogleAPIKey(“YOUR_GOOGLE_API_KEY”)
        //If you already have a mapControl running this needs to be destroyed
        mMapControl.onDestroy()
        mMapFragment.getMapAsync(this)
    }
}
```

</mi-tab-panel>
</mi-tabs>

## iOS

### Starting a Solution

> Just like Android, when you load your initial Solution, it's important to do it the "correct" way, to ensure it's easy to switch Solutions later if needed.

Reviewer 2: We can't assume the reader has read the Android section first.

```suggestion
When you load your initial Solution, it's beneficial to initialise MapsIndoors properly, to ensure it's easy to switch Solutions later if needed.
```

> > You don't have to initialise `positionProvider` or `enableLiveData`. This is only needed if you use third-party Positioning such as Cisco DNA Spaces, or MapsIndoors' LIve Data function.

Reviewer 2: ```suggestion
> You don't have to initialise `positionProvider` or `enableLiveData`. This is only needed if you use third-party Positioning such as Cisco DNA Spaces or MapsIndoors' Live Data function.
```

```swift
func setupMapIndoors(mapsIndoorsAPIKey: String, googleMapsAPIKey: String) {
    map = GMSMapView(frame: CGRect.zero)
    view = map
    mapControl = MPMapControl(map: map!)

    MapsIndoors.provideAPIKey(mapsIndoorsAPIKey, googleAPIKey: googleMapsAPIKey)
    MapsIndoors.synchronizeContent { error in
        // Orient your map to where you need data to be shown. This can e.g. be done by pointing the camera to a specific location or getting the default venue through MapsIndoors and panning the camera there
        self.map?.camera = .camera(withLatitude: 57.057964, longitude: 9.9504112, zoom: 20)

        // Setup needed services
        MapsIndoors.positionProvider = MyPositionProvider()
        MapsIndoors.positionProvider?.startPositioning(nil)
        self.mapControl?.enableLiveData(MPLiveDomainType.occupancy, handler: self)
    }
}
```

### Switching Solutions

To switch Solutions, you first need to ensure that all your existing instances are closed down safely. Once this is done, you can use `setupMapIndoors` to restart with your new desired API key.

> We recommend initialising your own function to call in the future for this purpose, like the example here with `switchSolution()`:

Reviewer 2: ```suggestion
We recommend creating your own function to call in the future for this purpose, like the example here with `switchSolution()`:
```
